pkg/console/console.go

```diff
@@ -0,0 +1,79 @@
+// Package console
+// The purpose of this package is to funnel all CLI console output, ensuring POSIX compliant use of stderr and stdout.
+package console
+
+import (
+	"fmt"
+	"github.com/samber/lo"
+	"io"
+	"os"
+)
+
+var (
+	verbose           = lo.ToPtr(false)
+	stdout  io.Writer = os.Stdout
+	stderr  io.Writer = os.Stderr
+)
+
+type Options struct {
+	Verbose *bool
+	Stdout  io.Writer
+	Stderr  io.Writer
+}
+
+func Configure(opts *Options) {
+	if opts.Verbose != nil {
+		verbose = opts.Verbose
+	}
+
+	if opts.Stdout != nil {
+		stdout = opts.Stdout
+	}
+
+	if opts.Stderr != nil {
+		stderr = opts.Stderr
+	}
+}
+
+// Stdoutf formats according to a format specifier and writes the resulting string to stdout, swallowing any errors.
+// DO NOT USE Stdoutf for debug statements, warnings, notifications, or errors.
+// When an output format flag such as `-o json` is used, stdout should have a single parsable JSON object written to it, so that it can be piped and parsed by tools such as JQ.
+func Stdoutf(format string, a ...any) {
+	_, _ = fmt.Fprintf(stdout, format, a...)
```

Comment:
is there somewhere any restriction what we pass into these functions ? if the requested "global" format specifier is json, would it work with the formatting code from api_error.go ?

Reply:
I answered this question in a Slack thread but I will put it here too for the curious.

We shouldn't be worrying about formatting errors into JSON or YAML.
In general with CLIs I have observed (kubectl, aws cli <- to name 2 big ones) that errors are printed in plain text to stderr with non-0 exit codes
See this example from the aws cli (RED carrets == the previous command exited non-zero and green == previous command exited 0)

![image](https://github.com/armory-io/armory-cli/assets/711726/a7376dcc-901f-44a0-9d3a-0774f60c4db8)

:point_up: following this pattern makes error handling much cleaner as you don't need to worry about what format a user requested and the stderr output logs to github logs and terminals even if you have the cli in pipes or `$()`

such as


```
#!/usr/bin/env bash

set -e
set -o pipefail

DEPLOYMENT_ID=$(armory deploy start -f scratch/deployment.yaml -o json | jq -r ".deploymentId")
```